### ISSUE TITLE
font path not set correctly

### ISSUE THREAD

**Issue by Hurgotron** [ArcticaProject/nx-libs#1039]
After setting up a new remote x2go instance (using Devuan), the font path was set to
```
/usr/share/fonts/X11/Type1,/usr/share/fonts/X11/75dpi,/usr/share/fonts/X11/100dpi,built-ins
```
which was missing (xterm complained) /usr/share/fonts/X11/misc. Comparing wirth the old instance, it showed that a symlink from /usr/share/nx/fonts to /usr/share/fonts/X11 was missing. Creating this (and restarting) made the font path change to
```
/usr/share/nx/fonts/misc/,/usr/share/nx/fonts/Type1/,/usr/share/nx/fonts/75dpi/,/usr/share/nx/fonts/100dpi/,/usr/share/fonts/X11/Type1,/usr/share/fonts/X11/75dpi,/usr/share/fonts/X11/100dpi,/usr/share/nx/fonts/Type1,/usr/share/nx/fonts/75dpi,/usr/share/nx/fonts/100dpi,built-ins
```
which contains the needed elements.

- is this the right way to do this?
- why was the symlink missing? install scripts broken in general, or just failing for me?
- is there some other way to set the (default) font path?
- in the ubuntu bug tracker is a similar issue, see https://bugs.launchpad.net/ubuntu/+source/x2goserver/+bug/1891825
- man page says ```'The default font path is __default_font_path__'``` which seems broken
- as far as I know, xorg x11 manages to dynamically change the font path with fontconfig. While this may be out of scope for nx, the proper way of changing the default font path, and a restart being needed, should be documented, because expectations of users may be different.

**Comment by uli42** [comment]
Which manpage are you referring to?

**Comment by Hurgotron** [comment]
manpage of nxagent - see the Ubuntu bug tracker link. Same issue in my man pages

**Comment by uli42** [comment]
I have openend #1041 for manpage part of this.

**Comment by uli42** [comment]
On my debian (not devuan) system I have a package called nx-X11-common which contains the missing symlink, amongst other stuff:
```
.
/usr
/usr/share
/usr/share/doc
/usr/share/doc/nx-x11-common
/usr/share/doc/nx-x11-common/changelog.Debian.gz
/usr/share/doc/nx-x11-common/changelog.gz
/usr/share/doc/nx-x11-common/copyright
/usr/share/fonts
/usr/share/fonts/X11
/usr/share/nx
/usr/share/nx/SecurityPolicy
/usr/share/nx/X11
/usr/share/nx/X11/XErrorDB
/usr/share/nx/X11/Xcms.txt
/usr/share/nx/fonts
```

So can you please check if your package dependencies are working?

**Comment by Hurgotron** [comment]
On my system, the package doesn't contain the symlink.

dpkg -L nx-x11-common
/.
/usr
/usr/share
/usr/share/doc
/usr/share/doc/nx-x11-common
/usr/share/doc/nx-x11-common/changelog.Debian.gz
/usr/share/doc/nx-x11-common/changelog.gz
/usr/share/doc/nx-x11-common/copyright
/usr/share/nx
/usr/share/nx/SecurityPolicy
/usr/share/nx/X11
/usr/share/nx/X11/XErrorDB
/usr/share/nx/X11/Xcms.txt

dpkg -l nx-x11-common
ii  nx-x11-common  2:3.5.99.26-2 all          nx-X11 (common files)

apt-cache show nx-x11-common
Package: nx-x11-common
Version: 2:3.5.99.26-2
Installed-Size: 369
Maintainer: Debian Remote Maintainers <debian-remote@lists.debian.org>
Architecture: all
Breaks: libnx-x11 (<< 2:3.5.0.29-1~), nxagent (<< 2:3.5.0.29-1~), nxlibs (<= 3.5.1)
Homepage: https://github.com/ArcticaProject/nx-libs/
Description-en: nx-X11 (common files)
 NX is a software suite which implements very efficient
 compression of the X11 protocol. This increases performance when
 using X applications over a network, especially a slow one.
 .
 This package provides all runtime architecture-independent files for
 nx-X11 Xserver (aka nxagent).
Description-md5: 0f6a6249e30d3d81043f7dce2e77290d
Source: nx-libs
Multi-Arch: foreign
Section: x11
Priority: optional
Filename: pool/DEBIAN/main/n/nx-libs/nx-x11-common_3.5.99.26-2_all.deb
Size: 200588
MD5sum: 9ef325649ee48700f2133afd42de5db1
SHA256: 63ad31d6e593e00662550a68b3280df062253ece3c22a12f8ba338e0a1916c8d

So it seems to be an original Debian package (most Devuan packages are) which is broken / incomplete?

**Comment by uli42** [comment]
Yes, seems so.

Hurgotron ***@***.***> schrieb am Mo., 28. Feb. 2022, 22:12:

**Comment by uli42** [comment]
I have just build debian packages with current HEAD and which resulted in nx-x11-common including the link. So can you please open a bugreport @debian?

**Comment by Hurgotron** [comment]
Opened Debian bug https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1006662

**Comment by sunweaver** [comment]
I just uploaded a fix for this to nx-libs 3.5.99.27-1 (testing/unstable) and about to upload as nx-libs 3.5.99.26-5+deb12u1 (bookworm). The official DEBs were missing debian/nx-x11-common.links (which is present in this repo). So the issue was specific to the official Debian package.
